clar2wasm/src/wasm_generator.rs

```diff
@@ -54,11 +54,51 @@ pub struct WasmGenerator {
     pub(crate) maps_types: HashMap<ClarityName, (TypeSignature, TypeSignature)>,
 
     /// The locals for the current function.
-    pub(crate) bindings: HashMap<String, Vec<LocalId>>,
+    pub(crate) bindings: Bindings,
     /// Size of the current function's stack frame.
     frame_size: i32,
 }
 
+#[derive(Debug, Clone, Default)]
+pub(crate) struct Bindings(HashMap<ClarityName, InnerBindings>);
+
+#[derive(Debug, Clone)]
+struct InnerBindings {
+    locals: Vec<LocalId>,
+    trait_id: Option<TraitIdentifier>,
```

Comment:
maybe nit: store the full `ty: TypeSignature` (instead of `trait_id`) and move the _check if it's a trait_ logic in `get_trait_name`?
Currently we only use the binding type to check if it's a trait, but maybe we will need to do something similar to handle other cases 🤷

Reply:
I asked myself this question for quite some time. I went with only the trait identifier in the end to avoid an absurd number of clones.
It might be cleaner though to have the type. I will change that.

Reply:
Done in [20877cd](https://github.com/stacks-network/clarity-wasm/pull/483/commits/20877cd542f5804aac928fe30c48f5c836613234)